### DISCUSSION TITLE
am i doing smth wrong?

### DISCUSSION THREAD

**Discussion by loneymist** [jasonacox/tinytuya#203] [Q&A] [answered]
Tried to make a light switcher. 

`data = d.status()`
Works fine and gives the status of bulb correctrly.
Output:
`{'dps': {'20': True, '21': 'white', '22': 168, '23': 750, '24': '00d0003203de', '25': '000e0d0000000000000000c80000', '26': 0}}`

But when i try to perform an turn it on\off nothing happens.
`import tinytuya`
`d = tinytuya.OutletDevice('abc', 'abc', 'abc')`
`d.set_version(3.3)`
`data = d.status() `
`print(data)`
`d.turn_off()`
So in output i can see an data but nothing happens with the bulb.

Although, i tried to control the color of bulb via
`d.set_colour(255,0,0)`
But it looks like python cannot se an `set_colour` attribute
Output:
` File "c:\PCMNGR\lampa.py", line 6, in <module>
    d.set_colour(255,0,0)
AttributeError: 'OutletDevice' object has no attribute 'set_colour'`

**Comment by uzlonewolf** [comment, marked as answer]
`set_colour()` is only available with `BulbDevice`, not `OutletDevice`,  so change `d = tinytuya.OutletDevice('abc', 'abc', 'abc')` to `d = tinytuya.BulbDevice('abc', 'abc', 'abc')`.  As that is a "Type B" bulb device, it uses DPS 20 for on/off control instead of the default DPS 1; you need to either make the above `OutletDevice` -> `BulbDevice` change (since BulbDevice handles this for you) or pass  `switch=20` to turn_on/turn_off with `d.turn_off(switch=20)`

**Reply by loneymist** [reply]
Thanks a lot!
